src/include/ikarus/Grids/SimpleGrid/SimpleGridFactory.h

```diff
@@ -23,7 +23,7 @@ namespace Ikarus::Grid {
     using VertexCoordinateType = Eigen::Vector<double, dimensionworld>;
 
   public:
-    void insertElement(const Dune::GeometryType& type, const std::span<size_t> vertices);
+    void insertElement(const Dune::GeometryType& type, std::span<size_t> vertices);

```

Comment:
The naming of this method is inherited from the dune grid factory interface. This should allow us to use Dune grids. 
Furthermore, and maybe more important we can use [Dune::GmshReader](https://www.dune-project.org/doxygen/2.5.0/classDune_1_1GmshReader.html) to read in gmsh files. Thus, we should refrain renaming this function to support this functionality. 

Alternative:
For using Dune grids we can write an adapter class. But to use GmshReader we need to write an adapter class for our own grid if we change `insertElement` to something like `insertRootEntity`

Reply:
What do you think @ToWibb ?